### ISSUE TITLE
quickfill: Speed up direct pixel access

### ISSUE THREAD

**Issue by Equbuxu** [PixiEditor/PixiEditor#230]
Surface.GetSRGBPixel and Surface.SetSRGBPixel functions are fairly slow. We need a faster way to set pixels when many pixels need to be accessed at once. 
I haven't thoroughly researched this, but here are some thoughts: 

- The simplest way to implement this would be copying SkSurface's pixels into a float[], reading/modifying them, and if any modifications were made applying those pixels back to Surface.
- It's possibly a better idea to use SKImage or SKBitmap in place of float[].
- As a matter of fact, some or all SkSurfaces already have a backing buffer stored in ram (this is used to display the contents of SKSurface on a writeablebitmap by creating a surface that uses the already existing internal buffer of the writeablebitmap). If this applies to all SKSurfaces, it seems reasonable to use that buffer for pixel access.

Point for discussion: the pixels are stored as 4 float16s inside the SKSufaces, but PixiEditor works with SKColor which uses 4 bytes.

**Comment by CPKreuz** [comment]
Is there a Color struct with 4 float16s?

**Comment by Equbuxu** [comment]
> Is there a Color struct with 4 float16s?

Yes, it's called SKColorF or SKFColor or something like that. However, I'm not sure it's a good idea to use the float format in tools, might be better to use SKColors and convert back and fourth (skia can do it very fast)

**Comment by CPKreuz** [comment]
Doesn't ReadPixels have a option to just read the part of the image that you need

So in our case 
ReadPixels(new ImageInfo(1, 1), pointerToBuffer, bytesPerRow, x, y)

**Comment by flabbet** [comment]
Smart solution
